plugins/Organic/Organic.cpp

```diff
@@ -231,71 +231,66 @@ void OrganicInstrument::playNote( NotePlayHandle * _n,
 		auto oscs_l = std::array<Oscillator*, NUM_OSCILLATORS>{};
 		auto oscs_r = std::array<Oscillator*, NUM_OSCILLATORS>{};
 
-		_n->m_pluginData = new oscPtr;
+		auto newOsc = new oscPtr;
+		_n->m_pluginData = newOsc;
 
-		for( int i = m_numOscillators - 1; i >= 0; --i )
+		for (int i = m_numOscillators - 1; i >= 0; --i)
 		{
-			static_cast<oscPtr *>( _n->m_pluginData )->phaseOffsetLeft[i]
-				= rand() / (static_cast<float>(RAND_MAX) + 1.0f);
-			static_cast<oscPtr *>( _n->m_pluginData )->phaseOffsetRight[i]
-				= rand() / (static_cast<float>(RAND_MAX) + 1.0f);
+			// Oscillator phases are wrapped within the range [0, 1), so 0 == 1
+			// TODO C++23: Use std::nextafter(1.f, 0.f) since it will be constexpr
+			constexpr float MAX_RANDOM_PHASE = 0.999999940395355224609375f;
+			newOsc->phaseOffsetLeft[i] = fastRand(MAX_RANDOM_PHASE);
+			newOsc->phaseOffsetRight[i] = fastRand(MAX_RANDOM_PHASE);
```

Comment:
I don't really like this. Maybe we could have a `fastRandEx` function (or similar name) whose upper bound is exclusive?

Reply:
Implemented as of 0e4ee0e51079404a4151931e9c5093137b1cec12